Fix vertical document scroll

lorado opened the pull request:
A potential fix for #240

wangyi7099 commented:
Thanks for PR ! Maybe we should rollback all the built files?

lorado commented:
@wangyi7099 You are welcome!
I was not sure, how you handle build/release cycle.
I made changes to `src/mode/native/core.js` and  `src/mode/native/mixins/update-native.js`, and run `yarn build`, so all files in `dist` folder are autogenerated. Should I revert them? Where do you built then dist files?
I also run lint and tests, all tests passed.

wangyi7099 commented:
> passed

Hi! Only when we publish a release, we will build the all files. In order to make commit more clear, we should't commit the built files.

wangyi7099 commented on src/mode/native/mixins/update-native.js:
```diff
@@ -44,12 +44,14 @@ export default {
         const ov = this.css(dom, 'overflow') || '';
         if (/scroll|auto/.test(ov)) {
-          const { v, h } = this.getScrollProcess(dom);
+          const { v, h, isVerticalScrollable } = this.getScrollProcess(dom);
```
Maybe we should also judge for the circumstance of horizontal scrolling?

lorado replied:
I was thinking about it, but I guess this is not really needed in real world. Mostly webpages are scroller vertically, so if you have a vuescroll component, that scrolls only vertically, but has same scrollWidth and clientWidth, it shouldn't be a problem, as the main document is also primary vertical scrollable, so the user will actually don't have a situation, where he tries to scroll to the right, and id would work... The user don't have a need to scroll to the right, I guess... =)
what do you think?

wangyi7099 replied:
I think it is necessay. We should cover all edge case.  
Image you have two scrollbars and they are inner and outer scrollbars. Inner scroll only scroll vertial and inner scroll only scroll horizontal .   And this bug will cause outer scroll could not be scrolled when your mouse is on inner scroll.

lorado replied:
hmm, good point. I'll add horizontal check too

lorado commented:
>Hi! Only when we publish a release, we will build the all files. In order to make commit more clear, we should't commit the built files.

Ok, gonna revert it then

lorado commented:
Ok, horizontal check is also implemented.
I also realized, that my changes were working only if vuescroll was imported from `vuescroll/dist/vuescroll-native` directly, so I added same logic to the `mix/core.js` file.

wangyi7099 reviewed (changes requested):
Hi! Bro! I think we should add judgement for scrollable in `checkScrollable`  rather than `getScrollProcess`. Because you have to add judgement in two files. It's making code redundant.
Like this:
```js
checkScrollable: function checkScrollable(
        e,
        deltaX,
        deltaY
      ) /* istanbul ignore next */ {
        var scrollable = false;

        // check mouse point scrollable.
        var dom = e.target ? e.target : e;
        while (
          dom &&
          dom.nodeType == 1 &&
          dom !== this.scrollPanelElm.parentNode &&
          !/^BODY|HTML/.test(dom.nodeName)
        ) {
          var ov = this.css(dom, 'overflow') || '';
          var isScrollX = this.css(dom, 'overflowX') !== 'hidden';
          var isScrollY = this.css(dom, 'overflowY') !== 'hidden';
          if (/scroll|auto/.test(ov)) {
            var _getScrollProcess = this.getScrollProcess(dom),
              v = _getScrollProcess.v,
              h = _getScrollProcess.h;

            if (
              (isScrollX && ((deltaX < 0 && h > 0) || (deltaX > 0 && h < 1))) ||
              (isScrollY && ((deltaY < 0 && v > 0) || (deltaY > 0 && v < 1)))
            ) {
              scrollable = dom == this.scrollPanelElm;
              break;
            }
          }
          dom = dom.parentNode ? dom.parentNode : false;
        }

        return scrollable;
      },
```

lorado commented:
I definitely agree with you, didn't come to this idea, as I was reading code from the native-mod perspective, and looked at the whole codebase afterwards, not realizing possible code improvements =)

wangyi7099 commented:
LGTM
